optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/optional/score/MeetingSchedulingConstraintProvider.java

```diff
@@ -0,0 +1,224 @@
+                .filter(meetingAssignment -> meetingAssignment.getStartingTimeGrain() != null)
+                .join(MeetingAssignment.class,
+                        filtering((leftAssignment, rightAssignment) -> rightAssignment.getStartingTimeGrain() != null),
+                        filtering((leftAssignment,
+                                rightAssignment) -> rightAssignment.getMeeting().getId() < leftAssignment.getMeeting().getId()),
+                        filtering((leftAssignment,
+                                rightAssignment) -> leftAssignment.getMeeting() != rightAssignment.getMeeting()),
+                        filtering((leftAssignment, rightAssignment) -> leftAssignment.calculateOverlap(rightAssignment) != 0))
+                .penalizeConfigurable("Overlapping meetings");
+    }
+
+    // TODO: Unspecified bug marked in DRL
+    protected Constraint assignLargerRoomsFirst(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUnfiltered(MeetingAssignment.class)
+                .filter(meetingAssignment -> meetingAssignment.getRoom() != null)
+                .join(Room.class,
+                        lessThan(MeetingAssignment::getRoomCapacity, Room::getCapacity))
+                .penalizeConfigurable("Assign larger rooms first",
+                        (meetingAssignment, room) -> room.getCapacity() - meetingAssignment.getRoomCapacity());
+    }
+
+    protected Constraint roomStability(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(Attendance.class)
+                .join(Attendance.class,
+                        equal(Attendance::getPerson, Attendance::getPerson),
+                        filtering((leftAttendance,
+                                rightAttendance) -> leftAttendance.getMeeting() != rightAttendance.getMeeting()))
+                .join(MeetingAssignment.class,
+                        filtering((leftAttendance, rightAttendance,
+                                leftAssignment) -> leftAttendance.getMeeting() == leftAssignment.getMeeting()),
+                        filtering((leftAttendance, rightAttendance,
+                                leftAssignment) -> leftAssignment.getStartingTimeGrain() != null))
+                .join(MeetingAssignment.class,
+                        filtering((leftAttendance, rightAttendance, leftAssignment,
+                                rightAssignment) -> rightAttendance.getMeeting() == rightAssignment.getMeeting()),
+                        filtering((leftAttendance, rightAttendance, leftAssignment,
+                                rightAssignment) -> rightAssignment.getStartingTimeGrain() != null),
+                        filtering((leftAttendance, rightAttendance, leftAssignment,
+                                rightAssignment) -> leftAssignment.getStartingTimeGrain().getGrainIndex() < rightAssignment
+                                        .getStartingTimeGrain().getGrainIndex()),
```

Comment:
From the looks of it, this should be doable with `lessThan(...)`.

Reply:
Same as above, how to put lambda expression in `lessThan()`?